### PR TITLE
Included shortened footnotes in references added counts

### LINKED ISSUE #3276: Include shortened footnotes in references added counts
The 'references added' feature uses ORES feature data to compare the number of references before and after a given edit. Specifically, it uses `feature.wikitext.revision.ref_tags`. This excludes refs that use the `{{sfn}}` (shortened footnote) template, which will undercount some articles or count some edits as removing refs when they are changing the ref format.

ORES now includes the shortened footnote count (`feature.enwiki.revision.shortened_footnote_templates`), so we should add that in to our calculations.

### PR DESCRIPTION
**Pull request by ANURADHAJHA99**
Included shortened footnotes in references added counts by creating a new variable `WIKITEXT_SHORT_TEMPLATE` and   `WIKITEXT_REF_TAGS_SHORT_TEMPLATE`, where the latter is the sum of 'feature.wikitext.revision.ref_tags' and 'feature.enwiki.revision.shortened_footnote_templates'.

closes #3276

### CODE REVIEW

**Comment by ragesoss** [comment]
The build is breaking during migrations. It looks like it's throwing an error from something in `Revision`, although it's not obvious to me what is wrong from a quick glance at the diff. Please make sure the build is passing locally for you and also on travis before requesting a PR review.

**Comment by ANURADHAJHA99** [comment]
@khyatisoneji I am unable to understand why there is a migration error, can you please guide me through the process a bit in the following build fail? So, that I can improve the code.

**Comment by ANURADHAJHA99** [comment]
@khyatisoneji I am having build error, I followed the steps as required, but I am not able to understand. Kindly please guide me through the following issue.

File: app/models/wiki_content/revision.rb [hunk 1]
```diff
@@ -68,16 +68,18 @@ def plagiarism_report_link
...
-  WIKITEXT_REF_TAGS = 'feature.wikitext.revision.ref_tags'
-  WIKIDATA_REFERENCES = 'feature.len(<datasource.wikidatawiki.revision.references>)'
+  WIKITEXT_REF_TAGS = 'feature.wikitext.revision.ref_tags'
+  WIKIDATA_REFERENCES = 'feature.len(<datasource.wikidatawiki.revision.references>)'
+  WIKITEXT_SHORT_TEMPLATE = 'feature.enwiki.revision.shortened_footnote_templates'
+  WIKITEXT_REF_TAGS_SHORT_TEMPLATE = WIKITEXT_REF_TAGS + WIKITEXT_SHORT_TEMPLATE
 
   def ref_tags
-    features[WIKITEXT_REF_TAGS] || features[WIKIDATA_REFERENCES]
+    features[WIKITEXT_REF_TAGS_SHORT_TEMPLATE] || features[WIKIDATA_REFERENCES]
```
Comment (khyatisoneji) [inline app/models/wiki_content/revision.rb]: It should be features[WIKITEXT_REF_TAGS] + features[WIKITEXT_SHORT_TEMPLATE]

File: app/models/wiki_content/revision.rb [hunk 2]
```diff
@@ -68,16 +68,18 @@ def plagiarism_report_link
...
   end
 
   def ref_tags_previous
     return 0 if new_article
-    features_previous[WIKITEXT_REF_TAGS] || features_previous[WIKIDATA_REFERENCES]
+    features_previous[WIKITEXT_REF_TAGS_SHORT_TEMPLATE] || features_previous[WIKIDATA_REFERENCES]
```
Comment (khyatisoneji) [inline app/models/wiki_content/revision.rb]: The same thing here

Comment (khyatisoneji) [reply app/models/wiki_content/revision.rb]: You can run tests on your terminal to check errors before pushing the PR,
you can install rubocop and call `bundle exec rubocop`  to check for any warning or Offenses

Run `bundle exec` to run all the tests

**Comment by ANURADHAJHA99** [comment]
@khyatisoneji I improved upon the suggestion provided by you and completed the work, there were a couple of problems that I was going through which were generally related to spaces. I solved them and ran the command `bundle exec rubocop`, I did come across a lot of offenses initially but then solved most of them. Now I have 0 offenses. However, the tests are failing again, I am unable to understand the problem, Can you please provide me some insight. Sorry to reply late, I was working on the issue for a long time, Thank you very much for your help. I have also attached below the screenshot of a successful test run.
Screenshot from 2019-10-30 22-11-50

**Comment by khyatisoneji** [comment]
I suggest you to make another function just as it has for just ref_tags
Make one for shortened_footened_templates which either include the value
features[WIKITEXT_SHORTENED_TEMPLATES] || 0
similarly make the function for features_previous

And instead of adding the features[WIKITEXT_SHORTENED_TEMPLATES] directly
in ref_tags function call the function made

**Comment by ANURADHAJHA99** [comment]
@khyatisoneji I was working on the issue, following the guidelines provided by you.  Can you guide me if I am right or wrong? 
Screenshot from 2019-10-31 20-17-43
Thank you for the help

**Comment by ragesoss** [comment]
The tests show errors related to the new code.

The best way to approach this is probably to find an example of a Wikipedia article that has both footnotes and shortened footnotes, and write a test against the expected number of total references.

**Pull request closed without merging**